doc/sphinx/src/contributing.rst

```diff
@@ -50,6 +125,118 @@ and we'll engage with you to figure out how to proceed.
 Notes for Contribuors
 ---------------------------------------
 
+Some notes on style and code architecture
+``````````````````````````````````````````
+
+* ``singularity-eos`` is primarily designed to provide needed equation
+  of state functionality to continuum dynamics codes. It isn't
+  supposed to provide the most accurate or complete picture of thermal
+  or statistical physics. As such the project tries to limit
+  capabilities to this scope.
+
+* A major influence on code style and architecture is the
+  `ten rules for developing safety-critical code`_, by Gerard Holzmann.
+  Safety critical code is code that exists in a context where failure
+  implies serious harm. A flight controler on an airplane or
+  spacecraft or the microcontroller in a car are examples of
+  safety-critical contexts. ``singularity-eos`` is not safety-critical
+  but many of the coding habits advocated for by Holzmann produce
+  long-lived, easy to understand, easy to parse, and easy to maintain code.
+  And we take many of the rules to heart. Here are a few that are most
+  relevant to ``singularity-eos``. They have been adapted slightly to 
+  our context.
+
+    #. Avoid complex flow constructs such as gotos.
+
+    #. All loops must have fixed bounds. This prevents runaway
+       code. (Note this implies that as a general rule, one should use
+       ``for`` loops, not ``while`` loops. It also implies one should
+       keep recursion to a minimum.)
+
+    #. Heap memory allocation should only be performed at
+       initialization. Heap memory de-allocation should only be
+       performed at cleanup.
+
+    #. Restrict the length of functions to a single printed page.
+
+    #. Restrict the scope of data to the smallest possible.
+
+    #. Use the preprocessor sparingly.
+
+    #. Limit pointer use to a single dereference. Avoid pointers of
+       pointers when possible.
+
+    #. Be compiler warning aware. Try to address compiler warnings as
+       they come up.
+
+.. _ten rules for developing safety-critical code: http://web.eecs.umich.edu/~imarkov/10rules.pdf
+
+* ``singularity-eos`` is a modern C++ code
+  and both standard template library capabilities and template
+  metaprogramming are leveraged frequently. This can sometimes make
+  parsing the code difficult. If you see something you don't
+  understand, ask. It may be it can be refactored to be more simple or
+  better documented.
```

Comment:
Looks like some typos here

Reply:
I must have blindness to my own writing, since I don't see the typos. Can you point them out?

Reply:
Never mind! I think I read it out of order when I saw it last lol. My mistake!